src/MudletInstanceCoordinator.cpp

```diff
@@ -137,3 +160,41 @@ void MudletInstanceCoordinator::handleDisconnected()
         socket->deleteLater();
     }
 }
+
+// Telnet URI queue and forwarding for multi-instance support
+void MudletInstanceCoordinator::queueTelnetUri(const QString& uri)
+{
+    QMutexLocker locker(&mMutex);
+    mQueuedTelnetUri = uri;
+}
+
+// Forward telnet URI to an already running Mudlet instance
+// Returns true on success
+bool MudletInstanceCoordinator::forwardTelnetUriToRunningInstance()
+{
+    QMutexLocker locker(&mMutex);
+    if (mQueuedTelnetUri.isEmpty()) {
+        return false;
+    }
+   
+    QLocalSocket socket;
+    socket.connectToServer(mServerName);
+    
+    if (socket.waitForConnected(WAIT_FOR_RESPONSE_MS)) {
+        // Prepend "TELNET_URI:" to distinguish from package paths
+        const QString message = qsl("TELNET_URI:") + mQueuedTelnetUri;
```

Comment:
`qsl(...)` is our macro (less to type) for `QStringLiteral(...)` and that can take replaceable arguments which is better than using `QString::operator+()`:
```suggestion
        const QString message = qsl("TELNET_URI:%1").arg(mQueuedTelnetUri);
```